tool/tsh/common/app_local_proxy.go

```diff
@@ -66,13 +67,35 @@ func newLocalProxyApp(tc *client.TeleportClient, appInfo *appInfo, port string,
 
 // newLocalProxyAppWithPortMapping creates a new generic app proxy. Unlike newLocalProxyApp, it
 // accepts a specific port mapping as an argument.
-func newLocalProxyAppWithPortMapping(tc *client.TeleportClient, appInfo *appInfo, portMapping client.PortMapping, insecure bool) *localProxyApp {
+func newLocalProxyAppWithPortMapping(ctx context.Context, tc *client.TeleportClient, appInfo *appInfo, app types.Application, portMapping client.PortMapping, insecure bool) (*localProxyApp, error) {
```

Comment:
I don't quite understand the idea behind `appInfo`. In theory, it has a `GetApp` method which lazily fetches the app or returns it if it was already fetched. However, to execute that method the callsite needs to have access to `api/client.GetResourcesClient`. Because of that, I opted for passing `types.Application` in addition to `appInfo`.

Reply:
You're right, `appInfo` isn't really useful here. We just need `proto.RouteToApp` and `profile` from it for some 
apps. I think we should remove it `localProxyApp` and add those two fields.

Alternatively, just set `appInfo.app = app` instead of adding a duplicate field.

Reply:
I replaced `appInfo` on `localProxyApp` with `profile` and `routeToApp`.